data_pipeline_api/registry/download.py

```diff
@@ -83,59 +84,71 @@ def _get_data_product_version_and_components(
-    data_products = get_data(
-        query_data, DataRegistryTarget.data_product, data_registry_url, token, exact=False
-    )
-    if data_products is None:
+    data_products_list = []
+    for ns in namespace:
+        query_data = {DataRegistryField.name: parsed_config.data_product, DataRegistryField.namespace: ns[DataRegistryField.url]}
+        if parsed_config.version is not None:
+            query_data[DataRegistryField.version] = parsed_config.version
+
+        # if query_data contains a version this will return a list with a single element, else it will return
+        # a list of all versions for the data_product
+        data_products = get_data(
+            query_data, DataRegistryTarget.data_product, data_registry_url, token, exact=False
+        )
+        if data_products:
+            data_products_list.extend(sort_by_semver(data_products))
+
+    if not data_products_list:
         raise ValueError(f"No data_product found with namespace {parsed_config.namespace}, name {parsed_config.data_product} and version {parsed_config.version}")
 
-    data_products = sort_by_semver(data_products)
-    if parsed_config.component:
-        data_product = None
-        data_product_components = []
-        # if a specific component has been requested then just get that component, iterate through the
-        # data_products, and for each get their component urls, get the data at that url and break on the
-        # first data_product found that contains the requested component
-        for dp in data_products:
-            obj = get_on_end_point(dp[DataRegistryField.object], token)
-            data_product_components = [
-                cn
-                for cn in [
-                    get_on_end_point(c, token)[DataRegistryField.name] for c in obj[DataRegistryField.components]
+    grouped_data_products = defaultdict(list)
+    for data_product in data_products_list:
+        grouped_data_products[tuple([data_product[DataRegistryField.namespace], data_product[DataRegistryField.name]])].append(data_product)
+
+    data_product_component_pairs = []
+    for _, group in grouped_data_products.items():
+        if parsed_config.component:
+            data_product = None
+            data_product_components = []
+            # if a specific component has been requested then just get that component, iterate through the
+            # data_products, and for each get their component urls, get the data at that url and break on the
+            # first data_product found that contains the requested component
+            for dp in group:
+                obj = get_on_end_point(dp[DataRegistryField.object], token)
+                data_product_components = [
+                    cn
+                    for cn in [
+                        get_on_end_point(c, token)[DataRegistryField.name] for c in obj[DataRegistryField.components]
+                    ]
+                    if re.match(glob.fnmatch.translate(parsed_config.component), cn)
```

Comment:
Why do you not just use `fnmatch.fnmatch`?

Reply:
I'd vaguely wondered if accessing it through glob made it a bit clearer about what was going on, but yeah I can just change it to fnmatch, probably that's fine